### PR TITLE
vmTools: set msize to 128KiB

### PR DESCRIPTION
**Pull request by Artturin**
500KiB was causing issues https://github.com/NixOS/nixpkgs/pull/142577

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/vm/default.nix [hunk 1]
```diff
@@ -1177,6 +1177,7 @@ rec {
   # The default 9P msize value is 8 KiB, which according to QEMU is
   # insufficient and would degrade performance.
   # See: https://wiki.qemu.org/Documentation/9psetup#msize
-  # Use 500 KiB as a conservative default, see also https://github.com/NixOS/nixpkgs/pull/142577#issuecomment-953848731
-  default9PMsizeBytes = 512000;
+  # Use 128KiB which is the default in linux 5.15+
+  # https://github.com/torvalds/linux/commit/9c4d94dc9a64426d2fa0255097a3a84f6ff2eebe
+  default9PMsizeBytes = 8 * 1024;
```
Comment (vcunat) [inline pkgs/build-support/vm/default.nix:1182]: I see 8k in here.  If it somehow means 128, I suppose the comment needs more explanation.

Comment (Artturin) [reply pkgs/build-support/vm/default.nix]: fixed

**Comment by Artturin** [comment]
currently taking times..

**Comment by talyz** [comment]
This still results in the same issue with the hibernate test. The `memorySize` still has to be increased.

**Comment by Artturin** [comment]
hibernate test fails with `virtualisation.memorySize = 1024;`
but with 
`virtualisation.memorySize = 2048;`
and adding 
```
documentation.enable = false;
documentation.nixos.enable = false;
```
(i noticed that the manual was being built when i changed the msize)
 
it takes



`2:26,41`(`135s`)  with`default9PMsizeBytes = 8 * 1024;`

`2:08,42`(`124s`) with `default9PMsizeBytes = 128 * 1024;`

**Comment by Artturin** [comment]
also, might be the time to change the default value of `virtualisation.memorySize` which has been 384M since 2009

**Comment by talyz** [comment]
Yep, 1024M seems like a more reasonable size today. A quick search for some different sizes revealed 37 instances of `memorySize` being changed to 1024.

**Comment by Artturin** [comment]
rg --pcre2 'memorySize' | rg --pcre2 '\d+(\d+)' --only-matching | sort | uniq -c
     43 1024
      4 128
      1 1300
      3 1536
      3 2000
      2 2046
     10 2047
     15 2048
      5 256
      1 3000
      1 386
      5 4096
      2 500
     10 512
      1 5120
      1 64
      1 686
      8 768
```

**Comment by talyz** [comment]
@GrahamcOfBorg test hibernate

**Comment by Artturin** [comment]
i'll make a separate pr bumping the default size and removing the explicit 1024 and less sizes

**Comment by Artturin** [comment]
i did a eval locally and ran the test too.

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/146804

**Comment by vcunat** [comment]
Closely related regression: https://github.com/NixOS/nixpkgs/pull/146810

**Comment by Artturin** [comment]
> Closely related regression: #146810

https://github.com/NixOS/nixpkgs/pull/146804#issuecomment-974715358

**Comment by Artturin** [comment]
another lowering https://github.com/NixOS/nixpkgs/pull/146816 hopefully this will be enough

**Pull request merged**